SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -292,13 +373,106 @@ void print_boxes_data(const std::vector<std::optional<Pokemon>>& boxes_data, Sin
     env.console.log(ss.str());
 }
 
+void output_boxes_data_json(const std::vector<std::optional<Pokemon>>& boxes_data, const std::string& json_path){
+    JsonArray pokemon_data;
+    for (size_t poke_nb = 0; poke_nb < boxes_data.size(); poke_nb++){
+        Cursor cursor = get_cursor(poke_nb);
+        JsonObject pokemon;
+        pokemon["index"] = poke_nb;
+        pokemon["box"] = cursor.box;
+        pokemon["row"] =  cursor.row;
+        pokemon["column"] =  cursor.column;
+        if (boxes_data[poke_nb] != std::nullopt){
+            // NOTE edit when adding new struct members
+            pokemon["national_dex_number"] =  boxes_data[poke_nb]->national_dex_number;
+            pokemon["shiny"] =  boxes_data[poke_nb]->shiny;
+            pokemon["gmax"] =  boxes_data[poke_nb]->gmax;
+            pokemon["ball_slug"] =  boxes_data[poke_nb]->ball_slug;
+            pokemon["gender"] = gender_to_string(boxes_data[poke_nb]->gender);
+            pokemon["ot_id"] = boxes_data[poke_nb]->ot_id;
+        }
+        pokemon_data.push_back(std::move(pokemon));
+    }
+    pokemon_data.dump(json_path);
```

Comment:
I would add the .json here. That way, we keep the right extension for the sorted version. The json is awesome